cli/src/infragpt/auth.py

```
@@ -0,0 +1,425 @@
+) -> None:
+    """Authenticate with InfraGPT platform using device flow."""
+    api_url = _get_api_base_url(api_base_url)
+    client = InfraGPTClient(api_base_url=api_url)
+
+    console.print("\n[bold]Authenticating with InfraGPT...[/bold]\n")
+
+    try:
+        flow = client.initiate_device_flow()
+    except InfraGPTAPIError as e:
+        console.print(f"[red]Error initiating authentication: {e.message}[/red]")
+        raise SystemExit(1)
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)